src/client/dfs/dfs.c

```diff
@@ -6887,77 +6953,100 @@ dfs_cont_check(daos_handle_t poh, const char *cont, uint64_t flags, const char *
-				entry.atime_nano = entry.mtime_nano = entry.ctime_nano =
-					now.tv_nsec;
-				entry.chunk_size = dfs->attr.da_chunk_size;
+	/** Start Pass 2 only if L+F flag is used */
+	if (!(flags & DFS_CHECK_RELINK))
+		goto done;
 
-				/*
-				 * Check if the oid corresponds to a file or dir using the oid type
-				 * and set the mode accordingly. symlink oids are never used/written
-				 * to since the value is stored in the entry itself and so will not
-				 * appear in this list.
-				 * Files are always arrays objects, so objects that are not arrays
-				 * have to be directories in this list.
-				 */
-				if (daos_is_array_type(otype)) {
-					entry.mode = S_IFREG | 0755;
-					/*
-					 * If this is a regular file / array object, the user might
-					 * have used a different chunk size than the default
-					 * one. Since we lost the directory entry where the chunks
-					 * size for that file would have been stored, we can make a
-					 * best attempt to set the chunk size. To do that, we get
-					 * the largest offset of every dkey in the array to see if
-					 * it's larger than the default chunk size. If it is, adjust
-					 * the chunk size to that, otherwise leave the chunk size as
-					 * default. This of course does not account for the fact
-					 * that if the chunk size is smaller than the default or is
-					 * larger than the largest offset seen then we have added or
-					 * removed existing holes from the file respectively. This
-					 * can be fixed later by the user by adjusting the chunk
-					 * size to the correct one they know using the daos tool.
-					 */
-					rc = adjust_chunk_size(dfs->coh, oids[i],
-							       &entry.chunk_size);
-					if (rc)
-						D_GOTO(out_lf2, rc);
-				} else {
-					entry.mode = S_IFDIR | 0755;
-				}
+	memset(&anchor, 0, sizeof(anchor));
+	while (!daos_anchor_is_eof(&anchor)) {
+		nr_entries = DFS_ITER_NR;
+		rc = daos_oit_list_unmarked(oit_args->oit, oids, &nr_entries, &anchor, NULL);
+		if (rc) {
+			D_ERROR("daos_oit_list_unmarked() failed: "DF_RC"\n", DP_RC(rc));
+			D_GOTO(out_lf2, rc = daos_der2errno(rc));
+		}
 
-				len = sprintf(oid_name, "%"PRIu64".%"PRIu64"",
-					      oids[i].hi, oids[i].lo);
-				D_ASSERT(len <= DFS_MAX_NAME);
-				rc = insert_entry(dfs->layout_v, now_dir->oh, DAOS_TX_NONE,
-						  oid_name, len, DAOS_COND_DKEY_INSERT, &entry);
-				if (rc) {
-					D_ERROR("Failed to insert leaked entry in l+f (%d)\n", rc);
-					D_GOTO(out_lf2, rc);
+		for (i = 0; i < nr_entries; i++) {
+			struct dfs_entry	entry = {0};
+			enum daos_otype_t	otype = daos_obj_id2type(oids[i]);
+			char			oid_name[DFS_MAX_NAME + 1];
+			daos_size_t		len;
+
+			/** in Pass 2 we should only see files */
+			if (!daos_is_array_type(otype)) {
+				D_ERROR("Directories should not be seen in Pass 2\n");
+				D_GOTO(out_lf2, rc = EIO);
+			}
+
+			if (flags & DFS_CHECK_PRINT)
+				D_PRINT("oid["DF_U64"]: "DF_OID"\n", unmarked_entries,
+					DP_OID(oids[i]));
```

Comment:
These files will be printed twice I assume, once for each pass?

Reply:
no because on the first pass we skip files if relink flag is passed. if not, we don't do the second pass.

Reply:
DFS_CHECK_RELINK and DFS_CHECK_PRINT can both be passed I assume?